src/t_list.c

```diff
@@ -99,12 +97,9 @@ static void listTypeTryConvertQuicklist(robj *o, int shrinking, beforeConvertCB
 
     /* Check the length or size of the quicklist is below the threshold. */
     int threshold = shrinking ? 2 : 1;
-    quicklistSizeAndCountLimit(server.list_max_listpack_size,&sz_limit,&count_limit);
-    if ((sz_limit != SIZE_MAX && ql->head->sz > sz_limit/threshold) ||
-        (count_limit != ULONG_MAX && ql->count > count_limit/threshold))
-    {
+    quicklistNodeLimit(server.list_max_listpack_size, &sz_limit, &count_limit);
+    if (ql->head->sz > sz_limit/threshold || ql->count > count_limit/threshold)
```

Comment:
p.s. that division bothered me for quite a while and i avoided commended since it's probably just my OCD.
but i imagine that the compiler uses division here, and if instead we'll do `if (shrinking) count_limit/=2, szlimit/=2;`, it'll just be two shifts.
feel free to ignore.

Reply:
Done